lib/vagrant-ovirt3/action/create_vm.rb

```diff
@@ -97,17 +97,20 @@ def call(env)
 
           # Wait till all volumes are ready.
           env[:ui].info(I18n.t("vagrant_ovirt3.wait_for_ready_vm"))
-          for i in 0..10
-            ready = true
-            server = env[:ovirt_compute].servers.get(env[:machine].id.to_s)
-            server.volumes.each do |volume|
-              if volume.status != 'ok'
-                ready = false
-                break
-              end
-            end
-            break if ready
-            sleep 2
+          for i in 0..20
+             ready = true
+             server = env[:ovirt_compute].servers.get(env[:machine].id.to_s)
+             server.volumes.each do |volume|
+               if volume.status != 'ok'
+                 ready = false
+                 break
+               end
+             end
+             if server.status != 'down'
```

Comment:
i'm not sure you want to do `server.status` inside of a `server.volumes.each`.

If you had 100 volumes you would check the `server.status` up to `20*100` times.